### PR TITLE
Change data save process to a two-step approach.

### PR DESCRIPTION
**Pull request by birgitlewicki**
Ticket: #680

### CODE REVIEW

File: src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        public dialogRef: MatDialogRef<DialogOkCancelComponent>,
+        @Inject(MAT_DIALOG_DATA) public data: DialogOkCancelData
+    ) {
+        this.dialogData = data;
+    }
+
+    cancelDialog() {
+        this.dialogRef.close(this.data.cancelText);
+    }
+
```
Comment (ruegenbfr) [inline src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts:30]: Please change function name
Please use Constants.DIALOG_OK rather then this.data.okText (to be more agnostic)

File: src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export class DialogOkCancelComponent {
+    dialogData: DialogOkCancelData;
+
+    constructor(
+        public dialogRef: MatDialogRef<DialogOkCancelComponent>,
+        @Inject(MAT_DIALOG_DATA) public data: DialogOkCancelData
+    ) {
+        this.dialogData = data;
+    }
+
```
Comment (ruegenbfr) [inline src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts:26]: Please use Constants.DIALOG_CANCEL rather then this.data.cancelText

**Review by ruegenbfr — APPROVED** [review]
The requested changes would be more suited within this pull request, but may be implemented in the next ticket.

File: src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts (hunk 1 above)
Comment (birgitlewicki) [reply src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts:30]: I done in Ticket #669

File: src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts (hunk 2 above)
Comment (birgitlewicki) [reply src/app/tracing/dialog/dialog-ok-cancel/dialog-ok-cancel.component.ts:26]: Is done in Ticket #669

**Pull request merged**